### PR TITLE
Add "implicit-dynamic: false" strong mode analyzer option support for generated files

### PR DESCRIPTION
**Pull request by caleo-git**

### CODE REVIEW

**Comment by kevmoo** [comment]
@caleo-git please rebase and squash. Also, you should update analysis options 

```
analyzer:
  strong-mode:
    implicit-casts: false
    implicit-dynamic: false
```
Or we're not going to be able to track this

**Comment by caleo-git** [comment]
@kevmoo I've enabled "implicit-dynamic: false" option, so now changes are ready for review

**Comment by caleo-git** [comment]
@kevmoo When you will be able to check this pull request?

File: json_serializable/lib/src/json_serializable_generator.dart [hunk 1]
```diff
@@ -94,7 +94,7 @@ class JsonSerializableGenerator
     // Explicitly using `LinkedHashMap` – we want these ordered.
     var fields = new LinkedHashMap<String, FieldElement>.fromIterable(
         fieldsList,
-        key: (f) => (f as FieldElement).name);
+        key: (FieldElement f) => f.name);
```
Comment (kevmoo) [inline json_serializable/lib/src/json_serializable_generator.dart]: Should be `key: (Object f) => (f as FieldElement).name);`

or you get https://travis-ci.org/dart-lang/json_serializable/jobs/308340768#L588

`uses_dynamic_as_bottom`

**Comment by kevmoo** [comment]
@jakemac53 – thoughts on this?

**Comment by nshahan** [comment]
@kevmoo I don't have much context here. In what settings are you getting the extra noise?

File: json_serializable/test/test_files/kitchen_sink.g.dart [hunk 2]
```diff
@@ -38,16 +38,12 @@ KitchenSink _$KitchenSinkFromJson(Map<String, dynamic> json) => new KitchenSink(
...
+              (dynamic e) => e == null ? null : DateTime.parse(e as String)))
   ..crazyComplex = (json['crazyComplex'] as List)
-      ?.map((e) => e == null
+      ?.map((dynamic e) => e == null
           ? null
-          : new Map<String, Map<String, List<List<DateTime>>>>.fromIterables(
-              (e as Map<String, dynamic>).keys,
-              (e as Map).values.map((e) =>
-                  e == null ? null : new Map<String, List<List<DateTime>>>.fromIterables((e as Map<String, dynamic>).keys, (e as Map).values.map((e) => (e as List)?.map((e) => (e as List)?.map((e) => e == null ? null : DateTime.parse(e as String))?.toList())?.toList())))))
+          : new Map<String, Map<String, List<List<DateTime>>>>.fromIterables((e as Map<String, dynamic>).keys, (e as Map).values.map((dynamic e) => e == null ? null : new Map<String, List<List<DateTime>>>.fromIterables((e as Map<String, dynamic>).keys, (e as Map).values.map((dynamic e) => (e as List)?.map((dynamic e) => (e as List)?.map((dynamic e) => e == null ? null : DateTime.parse(e as String))?.toList())?.toList())))))
```
Comment (jakemac53) [inline json_serializable/test/test_files/kitchen_sink.g.dart]: wow lol... presumably dartfmt is just giving up here?

Comment (kevmoo) [reply json_serializable/test/test_files/kitchen_sink.g.dart]: Yup!

**Comment by kevmoo** [comment]
@caleo-git – this adds a LOT of noise to the code.

How important is `implicit-dynamic: false` to you? Why are you setting it?

**Comment by caleo-git** [comment]
@kevmoo We are preparing migration to Dart 2.0 for our project. So we would like to have strong mode support for generated files.

**Comment by matanlurey** [comment]
@caleo-git `implicit-dynamic` has no (direct) connection for Dart 2.0 or strong-mode; it is an (optional) lint that doesn't have the full support of the Dart team at this time, and it would be very difficult to make guarantees that generated code will conform to it.

**Comment by kevmoo** [comment]
What @matanlurey said. Closing for now. Happy to set up time to talk about this on VC...

**Pull request closed without merging**